src/savegame_json.cpp

```diff
@@ -516,15 +516,13 @@ void Character::load( const JsonObject &data )
     } else {
         data.read( "mutations", my_mutations );
     }
-    for( auto it = my_mutations.begin(); it != my_mutations.end(); ) {
-        const auto &mid = it->first;
+    for( const trait_id mid : get_mutations() ) {
         if( mid.is_valid() ) {
             on_mutation_gain( mid );
             cached_mutations.push_back( &mid.obj() );
-            ++it;
         } else {
             debugmsg( "character %s has invalid mutation %s, it will be ignored", name, mid.c_str() );
-            my_mutations.erase( it++ );
+            my_mutations.erase( mid );
```

Comment:
This is not safe, of course neither was the previous code, so that's fun.
This needs to be like
```
for( auto it = my_mutations.begin(); it != my_mutations.end(); ) {
    const trait_id &mid = it->first;
    if( mid.is_valid() ) {
        // stuff
    } else {
        debugmsg( "ohnoes!" );
        it = my_mutations.erase( it );
```